src/dataflow/src/source/s3.rs

```diff
@@ -777,30 +771,30 @@ async fn download_object(
     };
 
     log::debug!(
-        "source_id={} {}/{} download_status={:?}",
+        "source_id={} {}/{} download_result={:?}",
         source_id,
         bucket,
         key,
-        download_status
+        download_result,
     );
 
-    if matches!(download_status, DownloadStatus::Ok) {
+    if let Ok(_) = &download_result {
```

Comment:
TIOLI (take it or leave it) (if I weren't asking for other stuff I wouldn't mention this): for things that don't actually care about the pattern matching properties I prefer to use either a method or the `matches!` macro:

```suggestion
    if download_result.is_ok() {
```